packages/web/src/components/cells/code.tsx

```diff
@@ -675,6 +677,44 @@ function tsLinter(
   });
 }
 
+function gotoDefinition(
+  pos: number,
+  cell: CodeCellType,
+  session: SessionType,
+  channel: SessionChannel,
+  editor: (content: string) => void,
+) {
+  async function gotoDefCallback({ response }: TsServerDefinitionLocationResponsePayloadType) {
+    channel.off('tsserver:cell:definition_location:response', gotoDefCallback);
+    if (response === null || response === undefined || Object.keys(response).length === 0) {
+      console.log('no response');
+      return;
+    }
+    const local: boolean =
+      response.file.includes('.srcbook/srcbooks') && !response.file.includes('node_modules');
```

Comment:
I assume these lines will not work on windows given the path separator being hardcoded? Can we have the server pass something down that is easier for the client to work with?

If the `TsServerDefinitionLocationResponsePayloadType` had a `cellId?: string` property on it, then we could simply do:

```typescript
async function gotoDefCallback(location: TsServerDefinitionLocationResponsePayloadType) {
  // ...
  if (location.cellId) {
    document.getElementById(`cell-${location.cellId}`)?.scrollIntoView({ behavior: 'smooth' });
  }
  //...
}
```

If `cellId` is not a string, then we know it needs a file look up.